xc/xc7/tests/soc/ibex/CMakeLists.txt

```diff
@@ -1,78 +1,107 @@
-add_file_target(FILE led.vmem)
-add_file_target(FILE pins_artya7.sdc)
+get_filename_component(IBEX_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../../../../../third_party/ibex ABSOLUTE)
+
+set(SV2V_ROM_INIT_FILE led.vmem)
+set(SV2V_IBEX_FLAGS --define=RegFile=ibex_pkg::RegFileFF --define=SRAMInitFile=${SV2V_ROM_INIT_FILE})
+
+# Get the list of source files used by the fusesoc to build the Ibex example
+
+execute_process(
+  COMMAND
+    bash generate.sh --only-deps
+  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+  OUTPUT_VARIABLE FILES_FROM_GENERATE_SCRIPT
+)
 
-get_target_property_required(PYTHON3 env PYTHON3)
+# Sort the file list into sources and includes
+
+set(SOURCE_FILE_LIST "")
+set(INCLUDE_FILE_LIST "")
+
+string(REPLACE "\n" ";" FILE_LIST "${FILES_FROM_GENERATE_SCRIPT}")
+foreach(FILE_NAME ${FILE_LIST})
+  get_filename_component(FILE_EXT ${FILE_NAME} EXT)
+  if (${FILE_EXT} STREQUAL ".sv")
+    list(APPEND SOURCE_FILE_LIST ${FILE_NAME})
+  elseif (${FILE_EXT} STREQUAL ".svh")
+    list(APPEND INCLUDE_FILE_LIST ${FILE_NAME})
+  endif()
+endforeach()
+
+# Add a command for generating the Ibex source files in the build directory
+
+add_custom_command(
+  OUTPUT ${SOURCE_FILE_LIST} ${INCLUDE_FILE_LIST}
+  COMMAND
+    bash generate.sh
+  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+  DEPENDS ${PYTHON3} ${IBEX_DIR}
+)
 
-# Arty
+# Get all the file targets for the includes and vmem files
+
+set(SOURCE_DEPS "")
+foreach(INC ${INCLUDE_FILE_LIST})
+  add_file_target(FILE ${INC} GENERATED)
+  get_file_target(TARGET ${INC})
+  list(APPEND SOURCE_DEPS ${TARGET})
```

Comment:
So the files from the `INCLUDE_FILE_LIST` should be added to the source file `INCLUDE_FILES` properties, and that should be all that is required here.  It will simplify these couple lines.

E.g. https://github.com/SymbiFlow/symbiflow-arch-defs/blob/b3864ae63d668fc7d321b94ca946841e80dd0bb6/common/cmake/file_targets.cmake#L293

Reply:
Done!